File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = prices[prices.length-1];
        int profit = 0;
        for(int i=prices.length-2;i>=0;i--){
            if(max-prices[i]>profit){
                profit=max-prices[i];
            }
            if(prices[i]>max){
                max = prices[i];
            }
        }
        return profit;
    }
}

File: Data Structures & Algorithms/last-stone-weight/submission-0.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones = [x*-1 for x in stones]
        heapq.heapify(stones)
        while len(stones)>1:
            stone1 = heapq.heappop(stones)
            stone2 = heapq.heappop(stones)
            if stone1!=stone2:
                max_stone = max(stone1,stone2)
                min_stone=min(stone1,stone2)
                heapq.heappush(stones,min_stone-max_stone)
        if len(stones)==0:
            return 0
        return -stones[0]